snapcraft/commands/remote.py

```diff
@@ -407,12 +372,95 @@ def _monitor_and_complete(  # noqa: PLR0912, PLR0915
         )
         return return_code
 
-    def _get_project_build_fors(self) -> list[str]:
-        """Get a unique list of build-for architectures from the project.
+    def _get_archs(self, build_fors: list[str], platforms: list[str]) -> list[str]:
+        """Get the architectures to build for.
+
+        If the project contains a ``platforms`` or ``architectures`` key, then project's
+        build plan is used to determine the architectures to build for. The build plan
+        can be filtered using the ``--platform`` or ``--build-for`` arguments.
+
+        If the project doesn't contain a ``platforms`` or ``architectures`` key, then
+        the architectures to build for are defined by the ``--platform`` or
+        ``--build-for`` arguments.
+
+        If there are no architectures defined in the project or as arguments, then the
+        default behavior is to build for the host architecture.
```

Comment:
```suggestion
        default behavior is to build for the host architecture of the local machine.
```

To disambiguate from the build host here.